Kitodo/src/main/java/org/kitodo/production/security/CustomLogoutSuccessHandler.java

```diff
@@ -41,12 +46,21 @@ public CustomLogoutSuccessHandler(String onSuccessUrl) {
     @Override
     public void onLogoutSuccess(HttpServletRequest request, HttpServletResponse response, Authentication authentication)
             throws IOException {
-        if (Objects.nonNull(authentication) && Objects.nonNull(authentication.getDetails())) {
-            Object principal = authentication.getPrincipal();
-            if (principal instanceof UserDetails) {
-                UserDetails user = (UserDetails) principal;
-                ServiceManager.getSessionService().expireSessionsOfUser(user);
+        if (Objects.nonNull(authentication)) {
+            if (Objects.nonNull(authentication.getDetails())) {
+                Object principal = authentication.getPrincipal();
+                if (principal instanceof UserDetails) {
+                    UserDetails user = (UserDetails) principal;
+                    ServiceManager.getSessionService().expireSessionsOfUser(user);
+                } else {
+                    logger.warn(MessageFormat.format("Cannot expire session: {0} !instanceof UserDetails",
```

Comment:
I am not a friend of this code like syntax in natual lanaguage logs. Can we change this to "{0}  is not an instance of 'UserDetails'" or something similar? (the `!` is easy to overlook in a log message, in my opinion)